glyphs-reader/src/plist.rs

```diff
@@ -472,6 +505,289 @@ impl From<BTreeMap<String, Plist>> for Plist {
     }
 }
 
+pub(crate) fn parse_int(s: &str) -> Result<i64, Error> {
+    if numeric_ok(s) {
+        if let Ok(num) = s.parse::<i64>() {
+            return Ok(num);
+        }
+        if let Ok(num) = s.parse::<f64>() {
+            return Ok(num as i64);
+        }
+    }
+    Err(Error::ExpectedNumber)
+}
+
+pub(crate) fn parse_float(s: &str) -> Result<f64, Error> {
+    if numeric_ok(s) {
+        if let Ok(num) = s.parse::<f64>() {
+            return Ok(num);
+        }
+    }
+    Err(Error::ExpectedNumber)
+}
+
+/// This type can be parsed from a Plist string
+pub trait FromPlist
+where
+    Self: Sized,
+{
+    fn parse(tokenizer: &mut Tokenizer<'_>) -> Result<Self, Error>;
+}
+
+impl<T> FromPlist for Vec<T>
+where
+    T: FromPlist,
+{
+    fn parse(tokenizer: &mut Tokenizer<'_>) -> Result<Self, crate::plist::Error> {
+        tokenizer.parse_delimited_vec(VecDelimiters::CSV_IN_PARENS)
+    }
+}
+
+impl<T> FromPlist for Option<T>
+where
+    T: FromPlist,
+{
+    fn parse(tokenizer: &mut Tokenizer<'_>) -> Result<Self, crate::plist::Error> {
+        Ok(Some(tokenizer.parse()?))
+    }
+}
+
+pub fn try_parse_plist<T>(plist: &str) -> Result<T, Error>
+where
+    T: FromPlist,
+{
+    Tokenizer::new(plist).parse()
+}
```

Comment:
An alternative to this would be to have an extra method on the `FromPlist` trait with an auto impl, like...

```rust
impl FromPlist: Sized {
    fn parse(tokenizer: &mut Tokenizer) -> Result<Self, crate::plist::Error> {
         ...
     }

    fn parse_plist_str(plist: &str) -> Result<Self, Error> {
        Tokenizer::new(plist).parse()
    }
}
```

And then you could just invoke it with, say, `String::parse_plist_str(&my_plist)`, without needing to think about another import.

Reply:
I like it, making it so